internal/common/common.go

```diff
@@ -437,3 +438,32 @@ func ApplyYamlPatch(src []byte, ops []byte) ([]byte, error) {
 
 	return patched, nil
 }
+
+func StrArrayToKernelArguments(args []string) (kargs models.KernelArguments) {
+	for _, s := range args {
+		kargs = append(kargs, &models.KernelArgument{
+			Operation: models.KernelArgumentOperationAppend,
+			Value:     s,
+		})
+	}
+	return
+}
+
+func UnmarshalKernelArguments(kargsStr string) (kargs models.KernelArguments, err error) {
+	if err = json.Unmarshal([]byte(kargsStr), &kargs); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal kernel arguments")
+	}
+	return
+}
+
+func KernelArgumentsToStrSlice(kargs models.KernelArguments) ([]string, error) {
+	var args []string
+	for _, arg := range kargs {
+		if arg.Operation != models.KernelArgumentOperationAppend {
+			return nil, errors.Errorf("only '%s' operation is allowed.  got '%s'", models.KernelArgumentOperationAppend,
+				arg.Operation)
+		}
+		args = append(args, arg.Value)
+	}
+	return args, nil
+}
```

Comment:
These are all only used in one place. And some are only used in tests.
That doesn't seem like they're generally useful enough to add to `common`.

Maybe if/when we end up with more callers this will make sense, but let's not add to `common` unless it's very necessary.

Reply:
Actually I wasn't I wasn't sure where is the best location to put them.
I wanted them to be together in a single place (not spread in several location), although their usage is from several files.
Where do you think is the best location?

Reply:
I think `StrArrayToKernelArguments` should be in `inventory_test.go` as that's the only place it's used.

I think `KernelArgumentsToStrSlice` and `UnmarshalKernelArguments` shouldn't be standalone functions at all.
Just put the content into `kernelArgsToSlice` directly as that's the only place they are used.